### ISSUE TITLE
New security rules (NULL check)

### ISSUE THREAD

**Issue by icy17** [the-tcpdump-group/libpcap#1336]
Hi, I've observed that some security rules are absent from the documentation. 

For the security rule "parameters must not be NULL," its omission in the documentation can lead to significant security issues.

Firstly, while best programming practices dictate that developers should avoid passing NULL parameters to an API, this guideline may not be obvious to less experienced developers, increasing the risk of overlooking necessary security checks.

Secondly, many APIs receive parameters from other functions or APIs, which might not be adequately checked for NULL, posing a risk even for experienced developers.

Therefore, it is crucial to explicitly mention in the documentation that NULL checks must be conducted on specified parameters before an API is used. Including these security rules will help developers identify potential security issues more effectively during testing.

I will provide a list of APIs requiring checks along with their corresponding parameters：

| API | parameter index (begin with 1) | 
|:----------:|:----------:|
| pcap_setfilter | 1 |
| pcap_next_ex | 1 |
| pcap_next_ex | 2 |
| pcap_open_offline | 2 |
| pcap_stats | 1 |
| pcap_get_selectable_fd | 1 |
| pcap_compile | 1 |
| pcap_compile | 2 |
| pcap_freecode | 1 |
| pcap_dump_close | 1 |
| pcap_dump_open | 1 |

**Comment by guyharris** [comment]
> Therefore, it is crucial to explicitly mention in the documentation that NULL checks must be conducted on specified parameters before an API is used.

For all those functions, the first argument to those functions is a `pcap_t *`.  The only way to get a valid `pcap_t *`is to call a routine such as `pcap_create()`, `pcap_open_offline()`, `pcap_open_live()`, etc.; in that case, the NULL check should be done after the call that *returns* a `pcap_t *`, which will ensure that no check need be done before each call.

**Comment by icy17** [comment]
> The only way to get a valid `pcap_t *`is to call a routine such as `pcap_create()`, `pcap_open_offline()`, `pcap_open_live()`, etc.; in that case, the NULL check should be done after the call that _returns_ a `pcap_t *`
I agree. It might be better to add a security rule to the API documentation to remind developers to check the return value.
